java/src/main/java/com/amido/stacks/workloads/menu/service/v1/MenuService.java

```diff
@@ -0,0 +1,51 @@
+package com.amido.stacks.workloads.menu.service.v1;
+
+import com.amido.stacks.workloads.menu.commands.CreateMenuCommand;
+import com.amido.stacks.workloads.menu.commands.UpdateMenuCommand;
+import com.amido.stacks.workloads.menu.domain.Menu;
+import com.amido.stacks.workloads.menu.exception.MenuAlreadyExistsException;
+import com.amido.stacks.workloads.menu.repository.MenuRepository;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MenuService {
+
+  protected MenuRepository menuRepository;
+
+  public Optional<Menu> create(Menu menu) {
+
+    menuRepository.save(menu);
+
+    return Optional.of(menu);
+  }
+
+  public void verifyMenuNotAlreadyExisting(CreateMenuCommand command) {
+    Page<Menu> existing =
+        menuRepository.findAllByRestaurantIdAndName(
+            command.getRestaurantId().toString(), command.getName(), PageRequest.of(0, 1));
+    if (!existing.getContent().isEmpty()
+        && existing.get().anyMatch(m -> m.getName().equals(command.getName()))) {
+      throw new MenuAlreadyExistsException(command, command.getRestaurantId(), command.getName());
+    }
+  }
+
+  public Optional<Menu> findById(String id) {
+    return menuRepository.findById(id);
+  }
+
+  public void delete(Menu menu) {
+    menuRepository.delete(menu);
+  }
+
+  public void update(Menu menu, UpdateMenuCommand command) {
```

Comment:
Probably worth returning Menu here, as that's what `save` returns so `return menuRepository.save()`